<launch>

    <arg name="project" default="GC_LOAM"/>
    
    <node pkg="tf" type="static_transform_publisher" name="base_link_joint" args="0 0 0 0 0 0 base_link chassis_link 100" />
    <node pkg="tf" type="static_transform_publisher" name="imu_joint" args="0 0 0 0 0 0 chassis_link imu_link 100" />
    <node pkg="tf" type="static_transform_publisher" name="gps_joint" args="0 0 0 0 0 0 chassis_link gps_link 100" />
    <node pkg="tf" type="static_transform_publisher" name="velodyne_joint" args="0 0 0 0 0 0 chassis_link velodyne 100" />

    <!-- Parameters -->
    <rosparam file="$(find GC_LOAM)/config/params.yaml" command="load" />

    <!--- LOAM -->
    <include file="$(find GC_LOAM)/launch/include/module_loam.launch" />
    <!-- <node pkg="$(arg project)" type="$(arg project)_featureExtract"   name="$(arg project)_featureExtract"    output="screen" 	respawn="true"/> -->

    <!--- Run Rviz-->
    <!-- <include file="$(find GC_LOAM)/launch/include/module_rviz.launch" /> -->
    <node pkg="rviz" type="rviz" name="$(arg project)_rviz" args="-d $(find GC_LOAM)/launch/include/rviz_lio_sam.rviz" />

    <!-- <arg name="bag_filename" default="/media/cc/A0EC52D0EC52A078/DATA/zhongche/2021-12-01-13-07-40.bag"/> -->
    <arg name="bag_filename" default="/media/cc/A0EC52D0EC52A078/DATA/VLP-16/2022-02-25-13-36-53.bag"/>
    <!-- <arg name="bag_filename" default="/media/cc/A0EC52D0EC52A078/DATA/wistron/new_driver/2021-09-28-14-43-19.bag"/> -->
    <node name="playbag" pkg="rosbag" type="play" args="--clock $(arg bag_filename) -s 80 -r 1.0 " />

</launch>
